M/Mineos/build_tarballs.jl

```diff
@@ -3,19 +3,37 @@
 using BinaryBuilder
 
 name = "Mineos"
-version = v"1.0"
+version = v"1.0.2" 
 
 # Collection of sources required to build Mineos
 sources = [
-    "https://github.com/anowacki/mineos.git" =>
-    "e2558b486d7656ef112608a8776643da66dc87cf",
+    GitSource("https://github.com/geodynamics/mineos.git", "3dd7c7433766d630b929d8254d03e705808ff8a3"),
+    DirectorySource("./bundled"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd ${WORKSPACE}/srcdir/mineos
-./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} --disable-doc
+
+# Drop docs build target
+atomic_patch -p1 ../patches/drop_docs.diff
+
+autoupdate
+autoreconf --install
+# Fix clang error 'error: non-void function * should return a value [-Wreturn-type]'
+if [[ "${target}" == *-freebsd* ]] || [[ "${target}" == *-apple-* ]]; then
+    export CFLAGS="-Wno-return-type"
+fi
+
+if [[ "${target}" == aarch64-apple-* ]]; then
```

Comment:
Is this not a property of the compiler rather than the architecture, or do I misunderstand?  I am aware that newer gfortran versions error by default when a scalar is passed in as the first element of a rank-1 array, but surely that is to do with the GCC version, not the architecture itself?  A different Fortran compiler on an Apple Silicon machine may behave differently, and GCC 10 on an x86 machine will throw the same error without the `-fallow-argument-mismatch` flag—as trying it on my Intel MacBook confirms.

Reply:
Digging deeper, it seems that `-fallow-argument-mismatch` has only been in gfortran since 2019, whereas the behaviour was previously implied by using the `-std=legacy` flag which still works in GCC 10.  Maybe that is a better flag to use if we know we are using gfortran, but don't want to add machinery to detect the version?  (Or perhaps we always use GCC >= 10 on Yggdrasil?)

Reply:
> Is this not a property of the compiler rather than the architecture, or do I misunderstand?

You're correct, but that's the only platform where we use GCC 11, and it's way easier to match target name rather than parsing the output of `gfortran --version` and comparing a version number

Reply:
Rearranged the comment so this is even more clear

Reply:
> that's the only platform where we use GCC 11, and it's way easier to match target name rather than parsing the output of gfortran --version and comparing a version number

Thanks for explaining that.  So I guess if and when the other architectures move on to GCC >= 10, any similar build scripts relying on `aarch64-apple` using a newer GCC will need to be rewritten.  Seems like unconditionally using `-std=legacy` might be easier, since there is no indication that that option is going away.

(`gfortran --version | awk 'NR == 1 {version = $NF; sub("\\..*", "", version); print version}'` seems easy enough.)

By the way, thank you @jeremiahpslewis for updating this!